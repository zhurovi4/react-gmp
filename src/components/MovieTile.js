import React from "react";
import PropTypes from "prop-types";
import styles from "./MovieTile.module.css";

const MovieTile = ({ movie, onClick }) => {
  const { imageUrl, name, releaseYear, genres } = movie;
  const genresString = genres.join(', ');

  return (
    <div className={styles.movieTile} onClick={() => onClick(movie)}>
      <img src={imageUrl} alt={name} />
      <div className={styles["movie-details"]}>
          <div>
            <div className={styles["movie-title"]}>{name}</div>
            <div className={styles["movie-genres"]}>
                  {genresString}
            </div>
          </div>
          <div className={styles["movie-year"]}>{releaseYear}</div>


      </div>
    </div>
  );
};

MovieTile.propTypes = {
  movie: PropTypes.shape({
    imageUrl: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    releaseYear: PropTypes.number.isRequired,
    genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  }).isRequired,
  onClick: PropTypes.func.isRequired,
};

export default MovieTile;
