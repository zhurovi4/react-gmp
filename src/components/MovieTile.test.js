import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import MovieTile from "./MovieTile";

describe("MovieTile component", () => {
  const movie = {
    imageUrl: "https://m.media-amazon.com/images/M/MV5BNDg2NjIxMDUyNF5BMl5BanBnXkFtZTgwMzEzNTE1NTM@._V1_FMjpg_UY508_.jpg",
    name: "Bohemian Rhapsody",
    releaseYear: 2021,
    genres: ["Biography", "Drama", "Music"],
  };
  const onClick = jest.fn();

  it("matches snapshot", () => {
    const { asFragment } = render(<MovieTile movie={movie} onClick={onClick} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders movie details", () => {
    render(<MovieTile movie={movie} onClick={onClick} />);
    expect(screen.getByText(movie.name)).toBeInTheDocument();
    expect(screen.getByText(`${movie.releaseYear}`)).toBeInTheDocument();
    expect(screen.getByText(`${movie.genres.join(", ")}`)).toBeInTheDocument();
  });

  it("calls onClick when clicked", () => {
    render(<MovieTile movie={movie} onClick={onClick} />);
    fireEvent.click(screen.getByRole("img"));
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith(movie);
  });
});
