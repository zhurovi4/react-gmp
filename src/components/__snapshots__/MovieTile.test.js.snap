// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="movieTile"
  >
    <img
      alt="Bohemian Rhapsody"
      src="https://m.media-amazon.com/images/M/MV5BNDg2NjIxMDUyNF5BMl5BanBnXkFtZTgwMzEzNTE1NTM@._V1_FMjpg_UY508_.jpg"
    />
    <div
      class="movie-details"
    >
      <div>
        <div
          class="movie-title"
        >
          Bohemian Rhapsody
        </div>
        <div
          class="movie-genres"
        >
          Biography, Drama, Music
        </div>
      </div>
      <div
        class="movie-year"
      >
        2021
      </div>
    </div>
  </div>
</DocumentFragment>
`;
